### ISSUE TITLE
B905 incorrectly raised on older python versions

### ISSUE THREAD

**Issue by danielpatrickdotdev** [astral-sh/ruff#6418]
In versions of python prior to 3.10, B905 "`zip()` without an explicit `strict=` parameter" shouldn't be raised. The parameter was not available in versions 3.9 and lower.

ruff version: `0.0.282`
command: `ruff check .`
minimal config: `select = ["B"]`
code example:

```python
for n, c in zip([0, 1, 2], ["A", "B", "C"]):
    print(n, c)
```

**Comment by charliermarsh** [comment]
I believe we have this gated correctly. Are you setting a [`target-version`](https://beta.ruff.rs/docs/settings/#target-version), or do you have a `requires-python` in your project? 

Example: `cargo run -p ruff_cli -- check foo.py --select B905 --target-version py38 --isolated` yields no errors on the above.

**Comment by charliermarsh** [comment]
(Closing as "working as intended", but happy to follow-up if you're still seeing issues when specifying a minimum supported version. We also changed 3.8 to the default minimum version in the latest release, which may resolve this for you.)

**Comment by danielpatrickdotdev** [comment]
Thank you for pointing me in the right direction to resolve this one!

I wouldn't have thought to try setting target-version as it's documented as "The minimum Python version to target". This application targets specifically 3.8 and no higher at present, and the issue with B905 is it's only relevant to _higher_ versions than 3.9.

I'm curious whether there is (or will be in future) any way for ruff to detect the python version without hardcoding it? We'll now be defining it in three places in three different formats - once for pyenv, once for poetry, and now also for ruff.

**Comment by zanieb** [comment]
@danielpatrickdotdev we'll infer `target-version` from `requires-python` in a `pyproject.toml` which is a standard central place to specify such things.

We can improve the documentation for `target-version` though.

> I'm curious whether there is (or will be in future) any way for ruff to detect the python version without hardcoding it?

I'm pretty interested in this but it increases the likelihood of different Ruff results for people working on a shared project.

**Comment by zanieb** [comment]
There are some caveats described in https://beta.ruff.rs/docs/settings/#target-version — are you using a `ruff.toml` file?
